source/docs/essentials/extend-the-graphql-schema.md

```diff
@@ -0,0 +1,443 @@
+
+```diff
+// my-module/server/modules/clicks-counters/index.js
++import typeDefs from "./schema.gql";
++
+export default {
+-  namespace: "ClicksCounters"
++  namespace: "ClicksCounters",
++  typeDefs: typeDefs
+};
+```
+
+<blockquote class="info">
+  Webpack is in fact taking care of converting SDL into a Javascript object
+  using an appropriate loader.
+</blockquote>
+
+**Congratulations again!** You should now be able to see these new fields and
+sue them in GraphQL.
+
+Try to execute the query below in your GraphQL playground (by default at
+[http://localhost:4000/playground](http://localhost:4000/playground)):
+
+```graphql
+# http://localhost:4000/playground
+{
+  product(sku: "WH09") {
+    sku
+    clicksCounter
+  }
+}
+```
+
+Our GraphQL module does not yet have any code for sending content, so you must
+see the following response:
+
+```json
+{
+  "data": {
+    "product": {
+      "sku": "WH09",
+      "clicksCounter": null
+    }
+  }
+}
+```
+
+Let’s now add some executable logic to our module.
+
+### Implement resolvers
+
+The GraphQL middleware relies on resolver functions to determine the data to
+return for a given field. This is where most of the « real work » is done, for
+instance by fetching remote datasources or transforming data.
+
+Resolvers are exposed using the `resolvers` key of the module definition. It
+should be a **Resolver map**: an object where each key is a GraphQL type name,
+and values are mapping between field names and resolver function. Resolver
+functions may return a
+[Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises)
+for asynchronous operations.
+
+<blockquote class="more">
+  To know more about resolvers and their internals, we recommend the reading of
+  [GraphQL Tools resolvers
+  documentation](https://www.apollographql.com/docs/graphql-tools/resolvers).
+</blockquote>
+
+First, update the module definition as follow:
+
+```diff
+// my-module/server/modules/clicks-counters/index.js
+import typeDefs from "./schema.gql";
++import resolvers from "./resolvers";
+
+export default {
+  namespace: "ClicksCounters",
+-  typeDefs: typeDefs
++  typeDefs: typeDefs,
++  resolvers: resolvers
+};
+```
+
+And then create the `my-module/server/modules/clicks-counters/resolvers.js` file
+with a resolver map as below (we are going to analyze it in details in a few
+seconds):
+
+```js
+// my-module/server/modules/clicks-counters/resolvers.js
+const counters = new Map();
+
+const currentValueOf = sku => counters.get(sku) || 0;
+
+module.exports = {
+  Product: {
+    clicksCounter: ({ sku }) => currentValueOf(sku)
+  },
+
+  Mutation: {
+    incrementProductCounter(_, { sku, incrementValue = 1 }) {
+      counters.set(sku, currentValueOf(sku) + incrementValue);
+      return {
+        success: true
+      };
+    }
+  }
+};
+```
+
+Let’s analyze this code part by part to understand what is possible to achieve
+in a module.
```

Comment:
I'm not sure about the module part. We only explain the resolvers here, not what a module is capable of. I'd simply remove it.